### PR TITLE
Update gitignore template

### PR DESCRIPTION
**Pull request by ameerthehacker**
After running the command 
```
npm run eject
```
If a git repo is setup now it includes unnecessary files from android and ios projects, thus I have updated the teplate with gitignore from the original react native repo

### CODE REVIEW

**Comment by ameerthehacker** [comment]
@gaving can anyone review my pull request

**Comment by evgenyrodionov** [comment]
Anyone?

**Comment by ameerthehacker** [comment]
hey @evgenyrodionov  i have been waiting for so long to get this reviewed

File: react-native-scripts/template/gitignore [hunk 1]
```diff
@@ -3,15 +3,71 @@
...
+
+# Node
+node_modules
+*.log
+.nvm
+/bots/node_modules/
+
+# TODO: Check in yarn.lock in open source. Right now we need to keep it out
+# from the GitHub repo as importing it might conflict with internal workspaces
+yarn.lock
```
Comment (brentvatne) [inline react-native-scripts/template/gitignore]: we should not be ignoring yarn.lock or package-lock.json, this seems like some weird thing internal to facebook

**Comment by brentvatne** [comment]
the correct gitignore to base this off of would be https://github.com/facebook/react-native/blob/master/local-cli/templates/HelloWorld/_gitignore

File: react-native-scripts/template/gitignore [hunk 2]
```diff
@@ -3,15 +3,71 @@
...
+
+# Watchman
+.watchmanconfig
+
+# Android
+.idea
+.gradle
+local.properties
+*.iml
+/android/
```
Comment (evgenyrodionov) [inline react-native-scripts/template/gitignore]: You will ignore entire `android` directory with `android/src` too

**Comment by ameerthehacker** [comment]
hey @evgenyrodionov, @brentvatne I have updated the gitignore as you suggested

**Comment by fson** [comment]
Thanks for the PR. The template has now moved to https://github.com/expo/new-project-template. Please re-open a PR there, if still relevant.

**Pull request closed without merging**